### PR TITLE
Fix in-memory analytics

### PR DESCRIPTION
**Pull request by grtlr**

### CODE REVIEW

File: src/db/collections/block.rs [hunk 1]
```diff
@@ -264,23 +265,35 @@ impl BlockCollection {
...
-                     metadata,
-                 }| (block_id, block, raw, metadata),
-            ))
+            .map_ok(|r| {
+                (
+                    r.block_id,
+                    RawMessage::<iota_types::block::Block>::from(r.raw.clone())
+                        .inner_unverified()
+                        .unwrap()
+                        .into(),
```
Comment (Alex6323) [inline src/db/collections/block.rs:292]: Can you help me understand this better? So ... eventually we need the protocol parameters from the time where this `Block` was serialized and written to the db?

File: src/db/collections/block.rs [hunk 2]
```diff
@@ -264,23 +265,35 @@ impl BlockCollection {
...
         index: MilestoneIndex,
     ) -> Result<impl Stream<Item = Result<(BlockId, Block, Vec<u8>, BlockMetadata), Error>>, Error> {
+        #[derive(Debug, Deserialize)]
+        struct QueryRes {
+            #[serde(rename = "_id")]
+            block_id: BlockId,
+            #[serde(with = "serde_bytes")]
+            raw: Vec<u8>,
+            metadata: BlockMetadata,
+        }
```
Comment (Alex6323) [inline src/db/collections/block.rs:275]: Why is this type necessary? Could we not just ignore `block` when mapping to our returned tuple? Or is this already the problem that the deserialization will fail?

File: src/db/collections/block.rs (hunk 1 above)
Comment (grtlr) [reply src/db/collections/block.rs:292]: No, the problem is we don't store `Output`s in the `BlockCollection`

File: src/db/collections/block.rs (hunk 2 above)
Comment (grtlr) [reply src/db/collections/block.rs:275]: Deserialization will fail. We should consider removing `derive(Deserialize)`.

Comment (Alexandcoats) [reply src/db/collections/block.rs:275]: It's unfortunate but we are allowed to gather the outputs by lookup, so it still needs to be `Deserialize`.

**Review by Alexandcoats — APPROVED** [review]

File: src/db/collections/block.rs (hunk 1 above)
Comment (Alex6323) [reply src/db/collections/block.rs:292]: ah, I see, thanks

**Review by Alex6323 — APPROVED** [review]

File: src/db/mongodb/mod.rs [hunk 3]
```diff
@@ -23,7 +23,8 @@ const DUPLICATE_KEY_CODE: i32 = 11000;
 /// A handle to the underlying `MongoDB` database.
 #[derive(Clone, Debug)]
 pub struct MongoDb {
-    pub(crate) db: mongodb::Database,
+    pub(crate) database_name: String,
+    //pub(crate) db: mongodb::Database,
```
Comment (Alexandcoats) [inline src/db/mongodb/mod.rs:27]: fmt

**Pull request merged**